hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -157,12 +160,19 @@ public void commitMigration(PartitionMigrationEvent event) {
     public void rollbackMigration(PartitionMigrationEvent event) {
         if (event.getMigrationEndpoint() == MigrationEndpoint.DESTINATION) {
             clearPartitionReplica(event.getPartitionId());
+        } else {
+            // MigrationEndpoint.DESTINATION
+            resetPartitionReplica(event.getPartitionId(), false);
         }
     }
 
     @Override
     public void clearPartitionReplica(int partitionId) {
-        segments[partitionId].clear();
+        resetPartitionReplica(partitionId, true);
+    }
+
+    private void resetPartitionReplica(int partitionId, boolean clear) {
```

Comment:
Only thing that bothers me is, names of `clearPartitionReplica` and `resetPartitionReplica` methods are sounds similar. It may be confusing.